internal/common/commonadapter/secret.go

```diff
@@ -139,3 +143,45 @@ func (s *SecretStore) Delete(ctx context.Context, secretID string) error {
 
 	return s.store.Delete(organizationID, secretID)
 }
+
+func (s *SecretStore) GetNameByID(ctx context.Context, secretID string) (string, error) {
+	organizationID, ok := s.extractor.GetOrganizationID(ctx)
+	if !ok {
+		return "", errors.NewWithDetails(
+			"organization ID cannot be found in the context",
+			"organizationId", organizationID,
+			"secretID", secretID,
+		)
+	}
+
+	secretResponse, err := s.store.Get(organizationID, secretID)
+	if err == secret.ErrSecretNotExists {
+		return "", errors.WrapIfWithDetails(err, "failed to get secret by ID", "secretID", secretID)
+	}
+	if err != nil {
+		return "", errors.WithDetails(errors.WithStackIf(err), "organizationID", organizationID, "secretID", secretID)
+	}
+
+	return secretResponse.Name, nil
+}
+
+func (s *SecretStore) GetSecretIDByName(ctx context.Context, secretName string) (string, error) {
+	organizationID, ok := s.extractor.GetOrganizationID(ctx)
```

Comment:
Should also try to extract organization ID from BRN secret ID. I'd create a helper method for org ID extraction and use that everywhere.

Reply:
i think it's unnecessary here, we are looking for `secretID` by secret name

Reply:
mea culpa